Hydro Tray lights now update immediately after pouring a reagent into it

DeityLink opened the pull request:
I am 100% sure it used to be like this 10 years ago but a decade of hydroponics reworks (~~both from bay and adam~~ not this time, it was all bay's fault) broke it at some point.

https://github.com/user-attachments/assets/83b9b3b8-6c78-4803-9820-57a7de64cd05

:cl:
* tweak: Hydro Tray lights now update immediately after pouring a reagent into it (reagents still get processed by the tray over time, but the lights anticipate how much the reagents currently held will change the tray's variables)
* tweak: Paint reagents are now slightly toxic to hydroponics plants, except flax oil which is slightly nutritious.
* tweak: Water and Holy Water now hydrate trays just as much
* tweak: Toxins in hydro tray and soil now become progressively more visible as it builds up, instead of suddenly becoming visible past a threshold.

Eneocho commented:
Tangentially related, but could you look into the proc speeds for reagents? Mutagen and radium, for example, tick mutations way slower than, say, bicaridine or hyperzine.

DeityLink commented:
I added optimizations to make sure that update_icon() doesn't get called more than it should, and fixed the toxins overlay looking wrong on soil, while also making it always appear, with its alpha depending on the current toxins level 
![image](https://github.com/user-attachments/assets/4a613cfa-f347-4f21-bbea-ec5181bbb130)
